### PR TITLE
Global configuration for deletion policy

### PR DESCRIPTION
**Pull request by MatejNedic**
AWS SQS: Add global configuration for deletionPolicy  #188

### CODE REVIEW

File: spring-cloud-aws-messaging/src/main/java/org/springframework/cloud/aws/messaging/listener/SqsMessageDeletionPolicy.java [hunk 1]
```diff
@@ -72,6 +73,12 @@
...
 	 * is thrown by the listener method, the message will not be deleted.
 	 */
-	ON_SUCCESS
+	ON_SUCCESS,
+
+	/**
+	 * If this is set in SqsListener, it will use global value set for specific QueueMessageHandlerFactory.
+	 * Default, if not changed is set to NO_REDRIVE.
+	 */
+	GLOBAL
```
Comment (maciejwalkowiak) [inline spring-cloud-aws-messaging/src/main/java/org/springframework/cloud/aws/messaging/listener/SqsMessageDeletionPolicy.java:82]: I think `DEFAULT` would be a better name. `global` -> `default` should be applied then in every other changed file.

File: spring-cloud-aws-messaging/src/main/java/org/springframework/cloud/aws/messaging/listener/QueueMessageHandler.java [hunk 2]
```diff
@@ -73,6 +74,7 @@
 	static final String LOGICAL_RESOURCE_ID = "LogicalResourceId";
 	static final String ACKNOWLEDGMENT = "Acknowledgment";
 	static final String VISIBILITY = "Visibility";
+	private  SqsMessageDeletionPolicy globalSqsMessageDeletionPolicy = SqsMessageDeletionPolicy.NO_REDRIVE;
```
Comment (maciejwalkowiak) [inline spring-cloud-aws-messaging/src/main/java/org/springframework/cloud/aws/messaging/listener/QueueMessageHandler.java:77]: Can we make this field as final and pass value in constructor? Regarding name `defaultMessageDeletionPolicy` would fit better.

File: spring-cloud-aws-messaging/src/main/java/org/springframework/cloud/aws/messaging/listener/SqsMessageDeletionPolicy.java (hunk 1 above)
Comment (MatejNedic) [reply spring-cloud-aws-messaging/src/main/java/org/springframework/cloud/aws/messaging/listener/SqsMessageDeletionPolicy.java]: True default is better. Done

File: spring-cloud-aws-messaging/src/main/java/org/springframework/cloud/aws/messaging/listener/QueueMessageHandler.java (hunk 2 above)
Comment (MatejNedic) [reply spring-cloud-aws-messaging/src/main/java/org/springframework/cloud/aws/messaging/listener/QueueMessageHandler.java]: Done

File: spring-cloud-aws-messaging/src/main/java/org/springframework/cloud/aws/messaging/listener/QueueMessageHandler.java [hunk 3]
```diff
@@ -73,15 +74,18 @@
 	static final String LOGICAL_RESOURCE_ID = "LogicalResourceId";
 	static final String ACKNOWLEDGMENT = "Acknowledgment";
 	static final String VISIBILITY = "Visibility";
+	private final   SqsMessageDeletionPolicy defaultSqsMessageDeletionPolicy;
```
Comment (eddumelendez) [inline spring-cloud-aws-messaging/src/main/java/org/springframework/cloud/aws/messaging/listener/QueueMessageHandler.java:77]: remove extra space. Also, can be renamed to `sqsMessageDeletionPolicy `

File: spring-cloud-aws-messaging/src/test/java/org/springframework/cloud/aws/messaging/config/annotation/SqsConfigurationTest.java [hunk 4]
```diff
@@ -119,6 +121,9 @@ void messageHandler_withFactoryConfiguration_shouldUseCustomValues()
 		assertThat(messageHandler.getCustomReturnValueHandlers().get(0)).isEqualTo(
 				ConfigurationWithCustomizedMessageHandler.CUSTOM_RETURN_VALUE_HANDLER);
 
+		Object sqsMessageDeletionPolicy = ReflectionTestUtils.getField(messageHandler,"defaultSqsMessageDeletionPolicy");
```
Comment (eddumelendez) [inline spring-cloud-aws-messaging/src/test/java/org/springframework/cloud/aws/messaging/config/annotation/SqsConfigurationTest.java:124]: there should be a space before `"defaultSqsMessageDeletionPolicy"`

File: spring-cloud-aws-messaging/src/test/java/org/springframework/cloud/aws/messaging/config/annotation/SqsConfigurationTest.java [hunk 5]
```diff
@@ -133,6 +138,42 @@ void messageHandler_withFactoryConfiguration_shouldUseCustomValues()
...
+		assertThat(messageHandler.getCustomArgumentResolvers().size()).isEqualTo(1);
+		assertThat(messageHandler.getCustomArgumentResolvers().get(0)).isEqualTo(
+			ConfigurationWithCustomizedMessageHandlerGlobalDeletionPolicy.CUSTOM_ARGUMENT_RESOLVER);
+
+		assertThat(messageHandler.getCustomReturnValueHandlers().size()).isEqualTo(2);
+		assertThat(messageHandler.getCustomReturnValueHandlers().get(0)).isEqualTo(
+			ConfigurationWithCustomizedMessageHandlerGlobalDeletionPolicy.CUSTOM_RETURN_VALUE_HANDLER);
+
+
+		Object sqsMessageDeletionPolicy = ReflectionTestUtils.getField(messageHandler,"defaultSqsMessageDeletionPolicy");
```
Comment (eddumelendez) [inline spring-cloud-aws-messaging/src/test/java/org/springframework/cloud/aws/messaging/config/annotation/SqsConfigurationTest.java:161]: there should be a space before "defaultSqsMessageDeletionPolicy"

File: spring-cloud-aws-messaging/src/main/java/org/springframework/cloud/aws/messaging/config/QueueMessageHandlerFactory.java [hunk 6]
```diff
@@ -52,6 +54,8 @@
 
 	private ResourceIdResolver resourceIdResolver;
 
+	private SqsMessageDeletionPolicy defaultSqsMessageDeletionPolicy = SqsMessageDeletionPolicy.NO_REDRIVE;
```
Comment (eddumelendez) [inline spring-cloud-aws-messaging/src/main/java/org/springframework/cloud/aws/messaging/config/QueueMessageHandlerFactory.java:57]: just `sqsMessageDeletionPolicy `

File: spring-cloud-aws-messaging/src/main/java/org/springframework/cloud/aws/messaging/listener/QueueMessageHandler.java (hunk 3 above)
Comment (MatejNedic) [reply spring-cloud-aws-messaging/src/main/java/org/springframework/cloud/aws/messaging/listener/QueueMessageHandler.java]: Done

File: spring-cloud-aws-messaging/src/test/java/org/springframework/cloud/aws/messaging/config/annotation/SqsConfigurationTest.java (hunk 4 above)
Comment (MatejNedic) [reply spring-cloud-aws-messaging/src/test/java/org/springframework/cloud/aws/messaging/config/annotation/SqsConfigurationTest.java]: Done

File: spring-cloud-aws-messaging/src/test/java/org/springframework/cloud/aws/messaging/config/annotation/SqsConfigurationTest.java (hunk 5 above)
Comment (MatejNedic) [reply spring-cloud-aws-messaging/src/test/java/org/springframework/cloud/aws/messaging/config/annotation/SqsConfigurationTest.java]: Done

File: spring-cloud-aws-messaging/src/main/java/org/springframework/cloud/aws/messaging/config/QueueMessageHandlerFactory.java (hunk 6 above)
Comment (MatejNedic) [reply spring-cloud-aws-messaging/src/main/java/org/springframework/cloud/aws/messaging/config/QueueMessageHandlerFactory.java]: Done

**Comment by maciejwalkowiak** [comment]
Thanks! I am merging it, don't worry about conflicts.

**Pull request closed without merging**